PyInstaller/depend/bytecode.py

```diff
@@ -81,30 +83,80 @@ def finditer(pattern: Pattern, string):
             break
 
 
-# language=PythonVerboseRegExp
-_call_function_bytecode = bytecode_regex(
-    rb"""
-    # Matches `global_function('some', 'constant', 'arguments')`.
-
-    # Load the global function. In code with >256 of names, this may require extended name references.
-    ((?:`EXTENDED_ARG`.)*
-     (?:`LOAD_NAME`|`LOAD_GLOBAL`|`LOAD_FAST`).)
-
-    # For foo.bar.whizz(), the above is the 'foo', below is the 'bar.whizz'.
-    ((?:(?:`EXTENDED_ARG`.)*
-     (?:`LOAD_METHOD`|`LOAD_ATTR`).)*)
-
-    # Load however many arguments it takes. These (for now) must all be constants.
-    # Again, code with >256 constants may need extended enumeration.
-    ((?:(?:`EXTENDED_ARG`.)*
-     `LOAD_CONST`.)*)
-
-    # Call the function. The parameter is the argument count (which may also be >256) if CALL_FUNCTION or CALL_METHOD
-    # are used. For CALL_FUNCTION_EX, the parameter are flags.
-    ((?:`EXTENDED_ARG`.)*
-     (?:`CALL_FUNCTION`|`CALL_METHOD`|`CALL_FUNCTION_EX`).)
-"""
-)
+if not compat.is_py311:
+    def _cleanup_code(code):
+        return code  # Nothing to do here
+
+    # language=PythonVerboseRegExp
+    _call_function_bytecode = bytecode_regex(
+        rb"""
+        # Matches `global_function('some', 'constant', 'arguments')`.
+
+        # Load the global function. In code with >256 of names, this may require extended name references.
+        ((?:`EXTENDED_ARG`.)*
+         (?:`LOAD_NAME`|`LOAD_GLOBAL`|`LOAD_FAST`).)
+
+        # For foo.bar.whizz(), the above is the 'foo', below is the 'bar.whizz'.
+        ((?:(?:`EXTENDED_ARG`.)*
+         (?:`LOAD_METHOD`|`LOAD_ATTR`).)*)
+
+        # Load however many arguments it takes. These (for now) must all be constants.
+        # Again, code with >256 constants may need extended enumeration.
+        ((?:(?:`EXTENDED_ARG`.)*
+         `LOAD_CONST`.)*)
+
+        # Call the function. The parameter is the argument count (which may also be >256) if CALL_FUNCTION or
+        # CALL_METHOD are used. For CALL_FUNCTION_EX, the parameter are flags.
+        ((?:`EXTENDED_ARG`.)*
+         (?:`CALL_FUNCTION`|`CALL_METHOD`|`CALL_FUNCTION_EX`).)
+    """
+    )
+else:
+    # Starting with python 3.11, the bytecode is peppered with CACHE instructions (which dis module conveniently hides
+    # unless show_caches=True is used). Dealing with these CACHE instructions in regex rules is going to render them
+    # unreadable, so instead we pre-process the bytecode and filter the offending opcodes out.
+    def _cleanup_code(code):
+        CACHE = dis.opmap["CACHE"]
+
+        out = []
+        for idx in range(0, len(code), 2):
+            if code[idx] == CACHE:
+                continue
+            out.append(code[idx])
+            out.append(code[idx+1])
+
+        return bytes(out)
```

Comment:
Would this just be too much regex :upside_down_face: ?
```suggestion
        return bytecode_regex(rb"(`CACHE`.)|(..)").sub(rb"\2", code)
```

Reply:
If it accounts for the fact that instruction occurs only on even bytes, then I'm all for regex...

Reply:
It does indeed! Although I'd prefer to see the `bytecode_regex(rb"(`CACHE`.)|(..)")` pulled out of the function so that it isn't calculated every time.

Reply:
Ah, neat! Because we always match two characters, the comparisons are indeed correctly aligned.